bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py

```diff
@@ -0,0 +1,277 @@
+    2. rdf.Filter(lambda x,y: x>y)
+        Filter using a python callable.
+    3. rdf.Filter(lambda x,y: x>y, ["y", "x"])
+        Filter using a python callable with column list passed explicitly.
+        When the column list is passed, the function maps the starting input parameters to those columns instead.
+        Here the function param x is mapped to column y, and the second function param y is mapped to column x.
+        This is equivalent to the filter "y>x".
+    4. rdf.Filter(lambda col1, const1: col1>const1, ["x"], extra_args = {"const1":0.5})
+        Filter using a python callable with column list passed explicitly and extra_args.
+        Here the function parameter col1 is mapped to col "x" of the RDF. and the const1 parameter is mapped to the float value 0.5
+        This is equivalent to the filter x>0.5"
+
+    Example:
+    rdf = ROOT.RDataFrame(5).Define("x", "(double) rdfentry_")
+    def x_greater_than_2(x): 
+        return x>2
+    fj = FunctionJitter(rdf)
+    func_call = fj.jit(x_greater_than_2)
+    fil1 = rdf.Filter( "Numba::" + func_call, "x is greater than 2")
+
+    """
+    if isinstance(callable_or_str, str): # If string argument is passed. Invoke the Original Filters.
+        return rdf._OriginalFilter(callable_or_str, *args)
+
+    if not callable(callable_or_str): # The 1st argument is either a string or a python callable.
+        raise TypeError(f"The first argument of a Filter operation should be a callable. {type(callable_or_str).__name__} object is not callable.")
+    
+    if len(args) > 2: 
+        raise TypeError(f"Filter takes at most 3 positional arguments but {len(args) + 1} was given")
+    
+    # Check if it is a c++ callable.
+    import libcppyy
+    if type(callable_or_str) == libcppyy.CPPOverload:
+        return rdf._OriginalFilter(callable_or_str, *args)
+    if getattr(callable_or_str, 'target_type', None) == libcppyy.CPPOverload:
+        return rdf._OriginalFilter(callable_or_str(), *args)
+    
+    
+    func = callable_or_str
+    jitter = FunctionJitter(rdf)
+    func.__annotations__['return'] = 'bool' # return type for Filters is bool # Note: You can keep double and Filter still works.
+
+    col_list = []
+    filter_name  = ""
+    
+    if len(args) == 1:
+        if isinstance(args[0], list): 
+            col_list = args[0]
+        elif isinstance(args[0], str):
+            filter_name = args[0]
+        else:
+            raise ValueError(f"Argument should be either 'list' or 'str', not {type(args[0]).__name__}.")
+    
+    elif len(args) == 2:
+        if isinstance(args[0], list) and isinstance(args[1], str):
+            col_list = args[0] 
+            filter_name = args[1]
+        else:
+            raise ValueError(f"Arguments should be ('list', 'str',) not ({type(args[0]).__name__,type(args[1]).__name__}.")
+            
+    
+    func_call = jitter.jit_function(func, col_list, extra_args)
+    return rdf._OriginalFilter("Numba::" + func_call, filter_name)
```

Comment:
We said you'll change this in another PR? I mean the way the jitting is implemented, not to use "Numba::".

Reply:
Yes. Hopefully with the PR for define.